server/channels/app/platform/web_broadcast_hook.go

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package platform
+
+import "github.com/mattermost/mattermost/server/public/model"
+
+type BroadcastHook interface {
+	// ShouldProcess returns true if the BroadcastHook wants to make changes to the WebSocketEvent.
+	ShouldProcess(msg *model.WebSocketEvent, webConn *WebConn, args map[string]any) bool
+
+	// Process takes a WebSocketEvent and modifies it in some way. It is passed a shallow copy of the WebSocketEvent,
+	// so if any nested fields such as data are modified, those need to be done using methods such as AddWithCopy.
+	Process(msg *model.WebSocketEvent, webConn *WebConn, args map[string]any) *model.WebSocketEvent
+}
+
+func (h *Hub) runBroadcastHooks(msg *model.WebSocketEvent, webConn *WebConn, hookIDs []string, hookArgs []map[string]any) *model.WebSocketEvent {
+	if len(hookIDs) == 0 {
+		return msg
+	}
+
+	// Check first if any hooks want to make changes to the event
+	hasChanges := false
+
+	for i, hookID := range hookIDs {
+		hook := h.broadcastHooks[hookID]
+		args := hookArgs[i]
+		if hook == nil {
```

Comment:
Is this case worth logging? I'd expect it not to happen in practice.

Reply:
Yep, I'll add it just in case. Cases we expect not to happen seem like the best things to log if there's no way to check them at compile time

Reply:
Maybe we can reduce the log level, it doesn't seem to be an actionable item by the administrator. Thoughts?

Reply:
Makes sense. I'll drop it to warning for now, but if you think it should go lower, I can do that